packages/shared/src/components/modals/report/ReportModal.tsx

```diff
@@ -17,15 +19,17 @@ interface Props extends ModalProps {
 
 export const OTHER_KEY = 'OTHER';
 
-export function ReportModal({
+export function ReportModal<
+  T extends ReportReason | PostModerationReason = ReportReason,
```

Comment:
As comment above, better named reject then report

Reply:
The `ReportReason` exists for some other reason (pun intended 🤣). This is used for the post/comment/source report modal.

For the case of Squad approval, we use `PostModerationReason`.